astropy/coordinates/representation.py

```diff
@@ -758,6 +1012,8 @@ def transform(self, matrix):
                        [ 1.23205081, 1.59807621],
                        [ 3.        , 4.        ]] pc>
         """
+        self._raise_if_has_differentials('transform')
```

Comment:
Is this necessary for `transform`? I think for cartesian, it is reasonable to transform the differentials as well, no? (sort-of assumed so in comments on transformations)